src/backend/test.rs

```diff
@@ -676,6 +789,118 @@ mod tests {
+
+        backend.assert_buffer_lines([
+            "          ",
+            "          ",
+            "          ",
+            "          ",
+            "          ",
+        ]);
+        backend.assert_scrollback_lines([
+            "aaaaaaaaaa",
+            "bbbbbbbbbb",
+            "cccccccccc",
+            "dddddddddd",
+            "eeeeeeeeee",
+            "          ",
+            "          ",
+            "          ",
+        ]);
+    }
+
+    #[test]
+    #[allow(deprecated)]
+    fn append_u16_max_plus_10_lines() {
+        let mut backend = TestBackend::new(10, 5);
+        backend.buffer = Buffer::empty(Rect::new(0, 0, 10, 5));
+
+        for i in 0..65546 {
+            let cells = format!("{i:>10}")
+                .drain(..)
+                .map(|c| {
+                    let mut cell = Cell::EMPTY;
+                    cell.set_char(c);
+                    cell
+                })
+                .collect::<Vec<_>>();
+            if i > 4 {
+                backend
+                    .set_cursor_position(Position { x: 0, y: 4 })
+                    .unwrap();
+                backend.append_lines(1).unwrap();
+            }
+            backend
+                .draw(
+                    cells
+                        .iter()
+                        .enumerate()
+                        .map(|(j, c)| (j as u16, 4.min(i) as u16, c)),
+                )
+                .unwrap();
+        }
+
+        backend.assert_buffer_lines([
+            "     65541",
+            "     65542",
+            "     65543",
+            "     65544",
+            "     65545",
+        ]);
+
+        // Make sure the scrollback is the right size.
+        assert_eq!(backend.scrollback.area.width, 10);
+        assert_eq!(backend.scrollback.area.height, 65535);
+        assert_eq!(backend.scrollback.content.len(), 10 * 65535);
+
+        // Compare the first 5 lines of the scrollback.
+        let mut scrollback_top = Buffer::empty(Rect::new(0, 0, 10, 5));
+        scrollback_top.content[0..10 * 5].clone_from_slice(&backend.scrollback.content[0..10 * 5]);
+        crate::assert_buffer_eq!(
+            &scrollback_top,
+            &Buffer::with_lines([
+                "         6",
+                "         7",
+                "         8",
+                "         9",
+                "        10",
+            ])
+        );
+
+        // Compare the last 5 lines of the scrollback.
+        let mut scrollback_bottom = Buffer::empty(Rect::new(0, 0, 10, 5));
+        scrollback_bottom.content[0..10 * 5]
+            .clone_from_slice(&backend.scrollback.content[10 * 65530..10 * 65535]);
+        crate::assert_buffer_eq!(
+            &scrollback_bottom,
+            &Buffer::with_lines([
+                "     65536",
+                "     65537",
+                "     65538",
+                "     65539",
+                "     65540",
+            ])
+        );
     }
```

Comment:
Do these simplifications make sense:
- removed the extraneous code
- moved the scrollback size checks after the content checks (as otherwise any test failures would just report just true/false results rather than 
- added a Cell::from(char) impl
- made a few parts a bit more concise

```suggestion
    #[test]
    fn append_lines_truncates_beyond_u16_max() -> io::Result<()> {
        let mut backend = TestBackend::new(10, 5);

        // Fill the scrollback with 65535 + 10 lines
        let row_count = u16::MAX as usize + 10;
        for row in 0..=row_count {
            if row > 4 {
                backend.set_cursor_position(Position { x: 0, y: 4 })?;
                backend.append_lines(1)?;
            }
            let cells = &format!("{row:>10}").chars().map(Cell::from).collect_vec();
            let content = cells
                .iter()
                .enumerate()
                .map(|(column, cell)| (column as u16, 4.min(row) as u16, cell));
            backend.draw(content)?;
        }

        // check that the buffer contains the last 5 lines appended
        backend.assert_buffer_lines([
            "     65541",
            "     65542",
            "     65543",
            "     65544",
            "     65545",
        ]);

        // TODO: ideally this should be something like:
        //     let lines = (6..=65545).map(|row| format!("{row:>10}"));
        //     backend.assert_scrollback_lines(lines);
        // but there's some truncation happening in Buffer::with_lines that needs to be fixed

        assert_eq!(
            Buffer {
                area: Rect::new(0, 0, 10, 5),
                content: backend.scrollback.content[0..10 * 5].to_vec(),
            },
            Buffer::with_lines([
                "         6",
                "         7",
                "         8",
                "         9",
                "        10",
            ]),
            "first 5 lines of scrollback should have been truncated"
        );

        assert_eq!(
            Buffer {
                area: Rect::new(0, 0, 10, 5),
                content: backend.scrollback.content[10 * 65530..10 * 65535].to_vec(),
            },
            Buffer::with_lines([
                "     65536",
                "     65537",
                "     65538",
                "     65539",
                "     65540",
            ]),
            "last 5 lines of scrollback should have been appended"
        );

        // These checks come after the content checks as otherwise we won't see the failing content
        // when these checks fail.

        // Make sure the scrollback is the right size.
        assert_eq!(backend.scrollback.area.width, 10);
        assert_eq!(backend.scrollback.area.height, 65535);
        assert_eq!(backend.scrollback.content.len(), 10 * 65535);

        Ok(())
    }
```

This requires a small change in Cell which I think is probably reasonable:
```
impl From<char> for Cell {
    fn from(ch: char) -> Self {
        let mut cell = Cell::EMPTY;
        cell.set_char(ch);
        cell
    }
}
```

These seem like nits, but for tests I like to be able to look at them and know that the test is right at a glance rather than have to process and predict how the test works before I can known that.